client/components/grade-form.jsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+
+export default class GradeForm extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      name: '',
+      course: '',
+      grade: ''
+    };
+
+  }
+
+  handleCancel(e) {
+    e.preventDefault();
+    this.setState({ name: '', course: '', grade: '' });
+  }
+
+  handleChange(e) {
+    this.setState({ [e.target.name]: e.target.value });
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    this.props.addStudent(this.state);
+    this.setState({ name: '', course: '', grade: '' });
+  }
+
+  render() {
+    return (
+      <form className="col-3" onSubmit={e => this.handleSubmit(e)}>
```

Comment:
In general, it's better to put column classes on containing boxes rather than on the component elements themselves. You don't want to mix layout with functionality if you can avoid it. Instead of making the form a `col-3`, put a `col-3` in App's render method and put your form component inside that.